File: Easy/9. Palindrome Number/9. Palindrome Number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0 || (x != 0 && x % 10 == 0)) {
            return false; // Negative numbers or multiples of 10 can't be palindromes
        }

        int original = x;
        int reversed = 0;

        while (x > 0) {
            reversed = reversed * 10 + (x % 10);
            x /= 10;
        }

        return original == reversed;
    }
}

File: Easy/9. Palindrome Number/9. Palindrome Number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x < 0 {
            return false; // Negative numbers can't be palindromes
        }

        let original = x;
        let mut reversed = 0;

        let mut x = x;
        while x > 0 {
            reversed = reversed * 10 + (x % 10);
            x /= 10;
        }

        original == reversed
    }
}
